src/containers/course-section/CourseSectionContainer.tsx

```diff
@@ -90,24 +95,42 @@ const CourseSectionContainer: FC<SectionProps> = ({
         ...sectionData.quizzes,
         ...sectionData.attachments
       ]
-      const updatedResourcesItems = prevResources.filter((prevResource) =>
-        allResourcesItems.some(
-          (currentResource) => currentResource._id === prevResource._id
+
```

Comment:
setResources function is too complex. I suggest splitting it into a couple of functions.

Reply:
Refactored setResources and splitted it into three smaller functions